packages/frontapp/src/app/components/BookCard/index.tsx

```diff
@@ -85,13 +94,47 @@ const BookCard: FC<IBookCardProps> = ({
   const handleItemRedirect = () => {
     navigate(`/item/${id}`);
   };
+  const [statusText, setStatusText] = useState('');
+  const currentStatus = getStatus(status?.status, date);
+  useEffect(() => {
+    console.log(currentStatus);
+    switch (currentStatus) {
+      case 'Free':
+        setStatusText('On the shelf');
+        break;
+      case 'Busy': {
+        const day = `${getDates(date).returnDate.getDate()}`.padStart(2, '0');
+        const month = `${getDates(date).returnDate.getMonth() + 1}`.padStart(
+          2,
+          '0'
+        );
+        setStatusText(`Return till: ${day}.${month}`);
+        break;
+      }
+      case 'Prolong': {
+        const day = `${getDates(date).returnDate.getDate()}`.padStart(2, '0');
+        const month = `${getDates(date).returnDate.getMonth() + 1}`.padStart(
+          2,
+          '0'
+        );
+        setStatusText(`Return till: ${day}.${month}`);
+        break;
+      }
+      case 'Overdue':
+        setStatusText('Overdue');
+        break;
+      default:
+        setStatusText('');
+        break;
+    }
+  }, [currentStatus]);
```

Comment:
I believe it causes unnecessary renders. This is one way function, so it might be used without `useEffect` and `setStatusText`. Just regular (input) => statusText